src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html

```diff
@@ -18,25 +25,25 @@
             <strong>{{ workItem.attributes['system.title'] }}</strong> with
           </label>
 
-          <div class="dib" dropdown id="iteration-select-dropdown">
-              <button #dropdownButton class="btn btn-default" type="button" dropdownToggle>
-                  {{ workItem.relationships.iteration?.attributes?.name || "Select iteration..." }} <span class="caret"></span>
-              </button>
-              <ul class="test" role="menu" dropdownMenu>
-                <li *ngFor="let iteration of iterations;" role="presentation">
-                  <a role="menuitem" tabindex="-1" (click)="onChangeIteration(iteration)">
-                    {{ iteration.attributes.name }}
-                  </a>
-                </li>
-              </ul>
+          <div class="parent-iteration-container input-group dropdown open active">
+            <input id="parent-iteration" class="form-control " type="text" #iterationSearch (keyup)="filterIteration($event)"
+            autocomplete="off" [value]="selectedIterationName" (focus)="iterationSearchFocus();">
+            <ul class="iteration-list dropdown-menu dropdown-ul" *ngIf="showIterationDropdown" #iterationList>
+              <li *ngIf="!filteredIterations.length">No matches found.</li>
+              <li *ngFor="let iteration of filteredIterations;" [attr.data-id]="iteration.key" (click)="associateIteration(iteration);">
+                <span [innerHTML]="(iteration.value) | almSearchHighlight: iterationSearch.value"></span>
+              </li>
+            </ul>
+            <span class="input-group-addon pointer" (click)="iterationSearchFocus();"> <span class="caret"></span> </span>
           </div>
         </div>
       </div>
 
       <!-- Action buttons to commit/abort -->
       <div class="text-right">
         <button class="btn" type="submit" id="cancel-iteration-button" (click)="iterationAssociationModal.close()">Cancel</button>
```

Comment:
this should be `<button class="btn btn-default">` and should not need custom styling

http://www.patternfly.org/pattern-library/widgets/#buttons